avm/ptn/deployment-script/import-image-to-acr/main.bicep

```diff
@@ -0,0 +1,255 @@
+metadata name = 'import-image-to-acr'
+metadata description = 'This modules deployes an image to an Azure Container Registry.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the deployment script resource.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// ============ //
+// Parameters   //
+// ============ //
+
+@description('Required. The name of the Azure Container Registry.')
+param acrName string
+
+@description('Optional. How the deployment script should be forced to execute. Default is to force the script to deploy the image to run every time.')
+param forceUpdateTag string = utcNow()
+
+@description('Optional. Azure RoleId that are required for the DeploymentScript resource to import images. Default is AcrPush, which is needed to import into an ACR.')
+// param rbacRoleNeeded string = '8311e382-0749-4cb8-b61a-304f252e45ec' // AcrPush
+param rbacRoleNeeded string = 'b24988ac-6180-42a0-ab88-20f7382dd24c' //Contributor is needed to import ACR
+
+@description('Optional. Does the Managed Identity already exists, or should be created. Default is false.')
+param useExistingManagedIdentity bool = false
```

Comment:
Can be removed if you just add a `managedIdentityResourceId` parameter that if empty means a managed identity must be created

Reply:
I didn't know the AVM module deployment-script existed :-( It makes sense to use it, you are absolutely right.